pkg/roachprod/roachprod.go

```diff
@@ -2511,3 +2497,19 @@ func CreateLoadBalancer(
 	}
 	return nil
 }
+
+func getCluster(l *logger.Logger, clusterName string) (*cloud.Cluster, error) {
+	// ListCloud may fail due to a transient provider error, but
+	// we may have still found the cluster we care about. It will
+	// fail below if it can't find the cluster.
+	cld, err := cloud.ListCloud(l, vm.ListOptions{})
+	c, ok := cld.Clusters[clusterName]
+	if !ok {
+		if err != nil {
+			return &cloud.Cluster{}, errors.Wrapf(err, "cluster not found")
+		}
+		return &cloud.Cluster{}, errors.New("cluster not found")
```

Comment:
`fmt.Errorf("cluster %s does not exist", clusterName)` since we know that `ListCloud` suceeded.

Reply:
Done.